### PR TITLE
Update storybook

### PR DESCRIPTION
**Pull request by jakemhiller**
Upgrades the storybook installation to the latest alpha.

This includes:

1. a new, much simpler, story format: [Component Story Format](https://storybook.js.org/docs/formats/component-story-format/)
2. A new "Docs" page format, to replace the old `info` plugin
3. Support for typescript types in the new format (Required some changes to gamut exports)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,6 +37,9 @@
     "packages/*",
     "workspaces/*"
   ],
+  "resolutions": {
+    "**/react-docgen": "5.0.0-beta.1"
```
Comment (jakemhiller) [inline package.json]: TODO: check if we still need this.

File: workspaces/gamut-storybook/.storybook/webpack.config.js [hunk 2]
```diff
@@ -66,6 +91,6 @@ module.exports = ({ config, mode }) => {
   }
 
   config.module.rules = config.module.rules.filter(Boolean);
-
-  return merge.smart(defaultConfig, config);
+  // console.log(config.module.rules[0].use[0].options.plugins);
```
Comment (jakemhiller) [inline workspaces/gamut-storybook/.storybook/webpack.config.js]:
```suggestion
```

File: packages/gamut/src/ButtonBase/index.tsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ export type ButtonBaseProps = (
   onClick?: (event: object) => void;
 };
 
-const ButtonBase = (props: ButtonBaseProps) => {
+export const ButtonBase = (props: ButtonBaseProps) => {
```
Comment (jakemhiller) [inline packages/gamut/src/ButtonBase/index.tsx]: Extracting props from storybook components requires a named export, not just a default. I think we want to start doing this by default moving forward, so it didn't seem like a big deal to add them alongside the default for now.

File: workspaces/gamut-storybook/.storybook/decorators/wrapper.js [hunk 4]
```diff
@@ -1,16 +1,9 @@
...
-  <Container
-    justify="stretch"
-    column
-    style={{ minHeight: '100vh', padding: '1rem' }}
-  >
-    {story()}
-  </Container>
-);
+export const wrapper = story => {
+  return <>{story()}</>;
```
Comment (JoshuaKGoldberg) [inline workspaces/gamut-storybook/.storybook/decorators/wrapper.js]: Curiosity: why not just...

```suggestion
  return story()
```

File: workspaces/gamut-storybook/.storybook/webpack.config.js [hunk 5]
```diff
@@ -43,21 +40,49 @@ module.exports = ({ config, mode }) => {
...
 
-  // Remove default storybook babel-loader
-  const babelIndex = config.module.rules.findIndex(
+  const jsIndex = config.module.rules.findIndex(
     r => r && r.test.test('test.js')
   );
-  if (babelIndex > -1) {
-    config.module.rules[babelIndex] = null;
+  if (jsIndex > -1) {
+    config.module.rules[jsIndex].test = /\.(mjs|(j|t)sx?)$/;
```
Comment (JoshuaKGoldberg) [inline workspaces/gamut-storybook/.storybook/webpack.config.js]: At this point, I've been so burned out modifying Webpack configs that this repeated object touching doesn't even bother me. 🙃

File: workspaces/gamut-storybook/typings.d.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+declare module 'global';
```
Comment (JoshuaKGoldberg) [inline workspaces/gamut-storybook/typings.d.ts]: 🤔 why is this necessary?

File: workspaces/gamut-storybook/.storybook/webpack.config.js (hunk 5 above)
Comment (jakemhiller) [reply workspaces/gamut-storybook/.storybook/webpack.config.js]: yeah, combining our config with the storybook config is a huge mess.

File: workspaces/gamut-storybook/typings.d.ts (hunk 6 above)
Comment (jakemhiller) [reply workspaces/gamut-storybook/typings.d.ts]: great question, i added this a while ago, might not be necessary now.

Comment (jakemhiller) [reply workspaces/gamut-storybook/typings.d.ts]: Yeah, there is actually a module called `global` that storybook depends on, and doesn't have any of its own types... at least as far as i can tell. Without this there are some compiler errors
